ee/app/livechat-enterprise/server/lib/VisitorInactivityMonitor.js

```diff
@@ -0,0 +1,65 @@
+import { SyncedCron } from 'meteor/littledata:synced-cron';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { settings } from '../../../../../app/settings/server';
+import { LivechatRooms, LivechatDepartment } from '../../../../../app/models/server';
+import { Livechat } from '../../../../../app/livechat/server/lib/Livechat';
+
+export class VisitorInactivityMonitor {
+	constructor() {
+		this._started = false;
+		this._name = 'Omnichannel Visitor Inactivity Monitor';
+	}
+
+	start() {
+		this._startMonitoring();
+	}
+
+	_startMonitoring() {
+		if (this.isRunning()) {
+			return;
+		}
+		const everyMinute = '* * * * *';
+		SyncedCron.add({
+			name: this._name,
+			schedule: (parser) => parser.cron(everyMinute),
+			job: () => {
+				this.handleAbandonedRooms();
+			},
+		});
+		this._started = true;
+	}
+
+	stop() {
+		if (!this.isRunning()) {
+			return;
+		}
+
+		SyncedCron.remove(this._name);
+
+		this._started = false;
+	}
+
+	isRunning() {
+		return this._started;
+	}
+
+	closeRooms(room) {
+		let comment = settings.get('Livechat_abandoned_rooms_closed_custom_message') || TAPi18n.__('Closed_automatically');
```

Comment:
How about creating a custom message `cache`(array), at the beginning of the `closeRooms` method and store the department custom messages there?
This would avoid unnecessary searches in the DB.